### ISSUE TITLE
WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files

### ISSUE THREAD

**Issue by Pitometsu** [NixOS/nixpkgs#16327]
Error output by Gtk applications.
```shell
 $ echo $XDG_CONFIG_DIRS
 
/home/netsu/.nix-profile/etc/xdg:/nix/var/nix/profiles/default/etc/xdg:/run/current-system/sw/etc/xdg

 $ tree /run/current-system/sw/etc/xdg/                                                  

/run/current-system/sw/etc/xdg/
├── autostart
│   ├── pulseaudio.desktop -> /nix/store/b8f56ikl1bvgg5va62p6cj7hgshjp6c9-pulseaudio-8.0/etc/xdg/autostart/pulseaudio.desktop
│   └── wicd-tray.desktop -> /nix/store/2ijqyvnms7j9h1fmn798g27igicm5p54-wicd-1.7.2.4/etc/xdg/autostart/wicd-tray.desktop
└── systemd -> /nix/store/d1w81dnv8wr55hwhwxdawwwx40f75ykx-systemd-230/etc/xdg/systemd

2 directories, 2 files

 $ tree /nix/store/0kq005z0cc5bfjais88ma92ycwrfwwxq-at-spi2-core-2.20.1/etc/xdg          

/nix/store/0kq005z0cc5bfjais88ma92ycwrfwwxq-at-spi2-core-2.20.1/etc/xdg
└── autostart
    └── at-spi-dbus-bus.desktop

1 directory, 1 file
```

**Comment by Pitometsu** [comment]
Could be solved by NO_AT_BRIDGE=1,  but this's temporary hack.

**Comment by Pitometsu** [comment]
Also error fixed if launch directly `/nix/store/0kq005z0cc5bfjais88ma92ycwrfwwxq-at-spi2-core-2.20.1/libexec/at-spi-bus-launcher`

**Comment by Pitometsu** [comment]
FYI: I don't use any DE, maybe this cause an issue.

**Comment by cstrahan** [comment]
I've seen similar problems running XMonad and no DE; would be interested in seeing what people suggest.

**Comment by Pitometsu** [comment]
Maybe I should enable `services.gnome3.at-spi2-core.enable`?

**Comment by DamienCassou** [comment]
I can reproduce with gnome 3.20:

```shellCommands
$ journalctl -b  | grep a11
juin 18 10:58:19 luz3 org.a11y.Bus[950]: GLib-GIO-Message: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
juin 18 10:58:19 luz3 org.a11y.Bus[950]: Activating service name='org.a11y.atspi.Registry'
juin 18 10:58:19 luz3 org.a11y.Bus[950]: ** (process:953): WARNING **: Failed to register client: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SessionManager was not provided by any .service files
juin 18 10:58:19 luz3 org.a11y.Bus[950]: Successfully activated service 'org.a11y.atspi.Registry'
juin 18 10:58:19 luz3 org.a11y.atspi.Registry[957]: SpiRegistry daemon is running with well-known name - org.a11y.atspi.Registry
juin 18 10:59:02 luz3 org.a11y.Bus[1201]: Activating service name='org.a11y.atspi.Registry'
juin 18 10:59:02 luz3 org.a11y.Bus[1201]: ** (process:1239): WARNING **: Failed to register client: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SessionManager was not provided by any .service files
juin 18 10:59:02 luz3 org.a11y.Bus[1201]: Successfully activated service 'org.a11y.atspi.Registry'
juin 18 10:59:02 luz3 org.a11y.atspi.Registry[1244]: SpiRegistry daemon is running with well-known name - org.a11y.atspi.Registry

```

**Comment by Pitometsu** [comment]
Yes, it solved issue.
***
So, in case of this error, here at least two options:

- to disable
```nix
variables = {
  NO_AT_BRIDGE = "1";
```

- to enable
```nix
services.gnome3.at-spi2-core.enable = true;
```

**Comment by Fabioamd87** [comment]
I solve this in ubuntu by installing 'at-spi2-core' package

**Comment by TekBoi** [comment]
> I solve this in ubuntu by installing 'at-spi2-core' package

This solved the issue for me as well running on Lubuntu 16.04

**Comment by scdas141** [comment]
Solved this issue for me as well, by installing 'at-spi2-core' package in Siduction 17.10 patience (based on Debian testing/unstable)..

Thanks...

**Comment by jtojnar** [comment]
Would it be a good idea to set `NO_AT_BRIDGE=1` in [at-spi2-core.nix](https://github.com/NixOS/nixpkgs/blob/release-17.03/nixos/modules/services/desktops/gnome3/at-spi2-core.nix) when the service is not enabled? Something like this:

```nix
  config = mkMerge [
    mkIf config.services.gnome3.at-spi2-core.enable {
      environment.systemPackages = [ pkgs.at_spi2_core ];

      services.dbus.packages = [ pkgs.at_spi2_core ];
    };

    mkIf !config.services.gnome3.at-spi2-core.enable {
      environment.variables.NO_AT_BRIDGE = 1;
    };
  ]
```

**Comment by abbradar** [comment]
It seems like a good idea to me; do you want to make a PR?

**Comment by jtojnar** [comment]
It still produces the warning for me, even with the service enabled.

**Comment by jtojnar** [comment]
The issue with missing systemd unit error will be fixed in #28856 

```
Error retrieving accessibility bus address: org.freedesktop.systemd1.NoSuchUnit: Unit at-spi-dbus-bus.service not found.
```

**Comment by Lew-Rockwell-Fan** [comment]
Me too. No, no, not *that*, I mean the warning message. As of March 2018 that was still popping up for me a lot in a plain Ubuntu 16.04 made from the mini.iso with x, lightdm, and openbox & no DE per se every time I used zenity for example. Installing at-spi2-core eliminated it.
